sylvia-derive/src/parser.rs

```diff
@@ -109,15 +109,20 @@ impl MsgType {
     pub fn emit_ctx_type(self) -> TokenStream {
         use MsgType::*;
 
+        let sylvia = crate_module();
+
         match self {
-            Exec | Instantiate => quote! {
-                (cosmwasm_std::DepsMut, cosmwasm_std::Env, cosmwasm_std::MessageInfo)
+            Exec => quote! {
+                #sylvia ::types::ExecCtx
+            },
+            Instantiate => quote! {
+                #sylvia ::types::InstantiateCtx
             },
             Migrate => quote! {
-                (cosmwasm_std::DepsMut, cosmwasm_std::Env)
+                #sylvia ::types::MigrateCtx,
             },
             Query => quote! {
-                (cosmwasm_std::Deps, cosmwasm_std::Env)
+                #sylvia ::types::QueryCtx
             },
```

Comment:
Here I am not exactly sure - we obviously want to support the new context types, but supporting original tuples or custom context types would also be nice. Iw ould prefer here to emit something like:

```
impl From<(#sylvia ::cw_std::DepsMut, #sylvia ::cw_std::Env, #sylvia ::cw_std::MessageInfo)>
```

I am not sure, but I think we use this in function arguments position, and this way you will be able to use any type constructible from tuple context. Obviously some `.into()` in generated could would have to be added.

Reply:
reverted to simple tuple